extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js

```diff
@@ -442,28 +477,56 @@ ve.init.mw.ViewPageTarget.prototype.loadAndRenderFancyCaptcha = function ( $cont
 	} );
 };
 
-/**
- * Get the licensing
- *
- * @returns {jQuery}
- */
-ve.init.wikia.ViewPageTarget.prototype.getLicense = function () {
-	if ( !this.$license ) {
-		this.$license = this.$('<div>')
-			.append(
-				this.$( '<p>' ).addClass( 've-ui-wikia-license' )
-					.html( ve.init.platform.getParsedMessage( 'copyrightwarning' ) )
-					.find( 'a' ).attr( 'target', '_blank' ).end()
-			);
+ve.init.wikia.ViewPageTarget.prototype.shouldReloadPageAfterSave = function () {
+	return this.userLoggedInDuringEdit ||
+		ve.init.wikia.ViewPageTarget.super.prototype.shouldReloadPageAfterSave.call( this );
+};
+
+ve.init.wikia.ViewPageTarget.prototype.getVeNotifyAfterSave = function () {
+	if ( this.restoring ) {
+		return 'restored';
+	} else if ( this.pageExists ) {
+		return 'saved';
+	} else {
+		return 'created';
 	}
-	return this.$license;
 };
 
-/**
- * Set up the license, attaching it after a surface.
- *
- * @param {ve.ui.Surface} surface Surface
- */
-ve.init.wikia.ViewPageTarget.prototype.setupLicense = function ( surface ) {
-	this.getLicense().insertAfter( surface.$element.closest( '.WikiaArticle' ) );
+ve.init.wikia.ViewPageTarget.prototype.updatePageOnCancel = function () {
+	if ( this.userLoggedInDuringEdit === true ) {
+		this.deactivating = true;
+		this.tearDownBeforeUnloadHandler();
+		// Reload the page so we don't need to worry about the user's state
+		window.location = this.viewUri;
+	} else {
+		ve.init.wikia.ViewPageTarget.super.prototype.updatePageOnCancel.call( this );
+	}
+};
+
+ve.init.wikia.ViewPageTarget.prototype.onLogInSuccess = function () {
+	var getEditTokenUrl = '/api.php?action=query&prop=info&titles=' +
```

Comment:
We could follow syntax that is used in: https://github.com/Wikia/app/blob/c9dce34f0717f98ef39712c6d2f06314f6dec19b/extensions/VisualEditor/modules/ve-mw/init/ve.init.mw.Target.js#L630-L630